### ISSUE TITLE
iOS's Low Power Mode causing downloads to hang

### ISSUE THREAD

**Issue by bitfabrikken** [birdofpreyru/react-native-fs#50]
When "Low Power Mode" is active, downloadFile times out.
Adjusting the timeouts does nothing - it never completes the download.
The built in js function, fetch(), works fine.

Reproduce:
- The component below or some code that uses RNFS.downloadFile.
- Enable Low Power Mode on the device: Settings > Battery > Low Power Mode
- Yank the power cable if it's charging - Low Power Mode is turned off internally even if it's enabled, when power is connected.

Caveats:
- Doesn't work on simulators (no Battery settings item to toggle it either)
- Some iOS versions are not affected, mostly iOS 14 or below are, but some devices on iOS 17 also time out. No pattern emerged it from my testing on 10ish devices. But seems the older the iOS, the more likely it'll be affected.

Reproduce code - remember to set device in Low Power Mode and unplug power. Plug in power to see it succeed with the download.
```js
import React, { Component } from 'react';

import * as RNFS from '@dr.pogodin/react-native-fs';

export default class App extends Component {

	constructor(props) {
		super(props);

		this.reproduce();
	}

	async reproduce(){

		var opts = {
			fromUrl: "https://picsum.photos/200/300",
			toFile: RNFS.DocumentDirectoryPath+"some_file_name",
			background: false,
			discretionary: false,
			cacheable: false,
			progressInterval: 500, 
			begin: (res)=>{ console.log('download begin')},
			progress: (res)=>{ console.log('progress',res)},
			resumable: false,
			connectionTimeout: 1000,
			readTimeout: 1000,
			backgroundTimeout: 1000,
		}

		try {
			var res = await RNFS.downloadFile(opts).promise;
		} catch (err) {
			//err here will be: [Error: The request timed out.]
			return alert("Error: "+err);
		}
		
		alert("download successful");
	}


	render(){
		return null;
	}
}
```

**Comment by birdofpreyru** [comment]
Hey @bitfabrikken , do you experience it with the latest library version? As the issue sounds awful similar to the problem https://github.com/birdofpreyru/react-native-fs/issues/24#issuecomment-2138066354, fixed in v2.26.0

**Comment by bitfabrikken** [comment]
@birdofpreyru, I just tested with 2.27.0, and it does indeed fix the problem! Thanks!
